doc/rst/technotes/foreach.rst

```diff
@@ -0,0 +1,38 @@
+.. _readme-foreach:
+
+====================
+The `foreach` Loop
+====================
+
+A ``foreach`` loop signifies that loop iterations are order-independent. This
+enables the compiler to vectorize the loop. In contrast, a ``for`` loop is
+order-dependent, and a ``forall`` loop can create parallel tasks and distribute
+iterations based on the iterator.
```

Comment:
Let's say explicitly here that a ``forall`` loop has order independent iterations

Reply:
that's if the iterator uses `foreach`, though, right?